openverse_catalog/dags/providers/provider_api_scripts/provider_data_ingester.py

```diff
@@ -0,0 +1,236 @@
+                should_continue = False
+
+        self.commit_records()
+
+    @abstractmethod
+    def get_next_query_params(self, old_query_params: Optional[Dict], **kwargs) -> Dict:
+        """
+        Given the last set of query params, return the query params
+        for the next request. Depending on the API, this may involve incrementing
+        an `offset` or `page` param, for example.
+
+        Required arguments:
+        old_query_params: Dictionary of query string params used in the previous
+                          request. If None, this is the first request.
+        **kwargs:         Optional kwargs passed through from `ingest_records`.
+
+        """
+        pass
+
+    def get_batch(self, query_params: Dict) -> Tuple[Optional[List], bool]:
+        """
+        Given query params, request the next batch of records from the API and
+        return them in a list.
+
+        Required Arguments:
+        query_params: query string parameters to be used in the request
+
+        Return:
+        batch:           list of records in the batch
+        should_continue: boolean indicating whether ingestion should continue after
+                         this batch has been processed
+        """
+        batch = None
+        should_continue = True
+
+        try:
+            # Get the API response
+            response_json = self.get_response_json(query_params)
+
+            # Build a list of records from the response
+            batch = self.get_batch_data(response_json)
+
+            # Optionally, apply some logic to the response to determine whether
+            # ingestion should continue or if should be short-circuited. By default
+            # this will return True and ingestion continues.
+            should_continue = self.get_should_continue(response_json)
+
+        except Exception as e:
+            logger.error(f"Error due to {e}")
+
+        return batch, should_continue
+
+    def get_response_json(self, query_params: Dict):
+        """
+        Make the actual API requests needed to ingest a batch. This can be overridden
+        in order to support APIs that require multiple requests, for example.
+        """
+        return self.delayed_requester.get_response_json(
+            self.endpoint, self.retries, query_params, headers=self.headers
+        )
+
+    def get_should_continue(self, response_json):
+        """
+        This method should be overridden when an API has additional logic for
+        determining whether ingestion should continue. For example, this
+        can be used to check for the existence of a `continue_token` in the
+        response.
+        """
+        return True
+
+    def get_batch_data(self, response_json):
+        """
+        Takes an API response and returns the list of records. This method
+        can be overridden to accomodate different response formats.
+        """
+        if response_json:
+            return response_json.get("data")
```

Comment:
It is a bit arbitrary -- there are a couple other providers with this response format, but it's far from common. I think you're right that this would be better to simply leave as abstract.